aws_lambda_powertools/event_handler/util.py

```diff
@@ -11,3 +16,27 @@ class _FrozenDict(dict):
 
     def __hash__(self):
         return hash(frozenset(self.keys()))
+
+
+def extract_origin_header(resolver_headers: Dict[str, Any]):
+    """
+    Extracts the 'origin' or 'Origin' header from the provided resolver headers.
+
+    The 'origin' or 'Origin' header can be either a single header or a multi-header.
+
+    Args:
+        resolver_headers (Dict): A dictionary containing the headers.
+
+    Returns:
+        Optional[str]: The value(s) of the origin header or None.
+    """
+    resolved_header = get_header_value(
+        headers=resolver_headers,
+        name="origin",
+        default_value=None,
```

Comment:
tip: you can set a same default like "". That way the caller doesn't need to check for None and instead always works with a string.

The same way you'd prefer to have an empty list or empty duct instead of optional list/dict, like the v3 for event source data classes.

Pre-approved

Reply:
In this case I think there is no problem in working with `None` because the function is already checking `None` values...

```python

def to_dict(self, origin: Optional[str]) -> Dict[str, str]:
        """Builds the configured Access-Control http headers"""

        # If there's no Origin, don't add any CORS headers
        if not origin:
            return {}
```

Reply:
It's less about being a problem but removing the need for checking for it and typing.

I'm gonna merge